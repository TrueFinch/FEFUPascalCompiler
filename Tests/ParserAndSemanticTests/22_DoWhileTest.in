program Project1;
var i: integer;
begin
  i := 0;
  do begin
    i += 1;
    write('2');
  end
  while (i < 2);
end.