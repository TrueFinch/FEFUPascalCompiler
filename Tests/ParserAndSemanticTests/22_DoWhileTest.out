                   project1                    
                      ┼                        
                  MainBlock                    
       ┌──────────────┴───────┐                
      var             CompoundStatement        
       ┼         ┌────────────┴──┐             
 SimpleVarDecl   :=      DoWhileStatement      
 ┌─────┼        ┌┴─┐   ┌─────────┴──┐          
 i  integer     i  0   <    CompoundStatement  
                      ┌┴─┐   ┌──────┼          
                      i  2   +=   write        
                            ┌┴─┐  ┌─┴          
                            i  1  2            